plugin/geoip/geoip.go

```diff
@@ -0,0 +1,102 @@
+// Package geoip implements a max mind database plugin.
+package geoip
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+	"github.com/oschwald/geoip2-golang"
+)
+
+var log = clog.NewWithPlugin(pluginName)
+
+// GeoIP is a plugin that add geo location data to the request context by looking up a maxmind
+// geoIP2 database, and which data can be later consumed by other middlewares.
+type GeoIP struct {
+	Next          plugin.Handler
+	db            db
+}
+
+type db struct {
+	*geoip2.Reader
+	// provides defines the schemas that can be obtained by querying this database, by using
+	// bitwise operations.
+	provides int
+}
+
+const (
+	anonymousIP = 1 << iota
+	asn
+	city
```

Comment:
only city is used from this list of constant
(at least `domain` is not used - which I what I checked)

Reply:
I felt it make sense to keep the entire list for forward compatibility in case we decide to use other database types in the future, and to match the existing list in the [geopip2-golang library](https://github.com/oschwald/geoip2-golang/blob/main/reader.go#L206-L215).

Having our own list with different values for different schemas would make things harder and may get out of sync with the original database type list of values.

WDYT?